ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/IntroEligibilityStateView.kt

```diff
@@ -0,0 +1,61 @@
+package com.revenuecat.purchases.ui.revenuecatui.composables
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+
+internal enum class IntroOfferEligibility {
+    INELIGIBLE,
+    ELIGIBLE,
+}
+
+/**
+ * A composable that can display different data based on intro eligibility status.
+ */
+@Composable
+internal fun IntroEligibilityStateView(
+    textWithNoIntroOffer: String?,
+    textWithIntroOffer: String?,
+    eligibility: IntroOfferEligibility,
```

Comment:
So I'm guessing the view model will calculate the `IntroOfferEligibility` from the presence of a free phase/intro phase in the subscription option right?

Reply:
Yup